### PR TITLE
Give a clear error message when adding package with the same name as your local package

### PR DESCRIPTION
**Pull request by david-fortini**
# Pull Request Check List

Resolves: #236

- [x] Added **tests** for changed code.
- [x] Updated **documentation** for changed code.

### CODE REVIEW

File: poetry/mixology/incompatibility.py [hunk 1]
```diff
@@ -57,7 +57,10 @@ def __init__(self, terms: List[Term], cause: IncompatibilityCause) -> None:
                     # irrelevant, since we already know that mutually exclusive version
                     # ranges are incompatible. We should never derive an irrelevant
                     # incompatibility.
-                    assert by_ref[ref] is not None
+                    err_msg = "Package '{}' is listed as a dependency of itself.\nThis is not supported by Poetry.".format(
+                        ref
+                    )
```
Comment (finswimmer) [inline poetry/mixology/incompatibility.py:62]:
```suggestion
                    err_msg = f"Package '{ref}' is listed as a dependency of itself."
```

I'm not aware that listing itself is possible at all. That's why I would remove this sentence.

**Review by finswimmer — CHANGES_REQUESTED** [review]
Thanks a lot for your contribution @david-fortini :+1: 

I've added just a small suggestion.

File: poetry/mixology/incompatibility.py (hunk 1 above)
Comment (david-fortini) [reply poetry/mixology/incompatibility.py:62]: Sure makes sense.

**Comment by hagaivcita** [comment]
hi, looks like the tests are passing except some CI procedures which are stuck (runs for 100+ days)
can they be reinitialized so this PR would merge and resolve?

**Comment by finswimmer** [comment]
Could you please rebase your PR @david-fortini ?

**Comment by david-fortini** [comment]
> Could you please rebase your PR @david-fortini ?

Yes I will do that in the coming weekend!

**Comment by david-fortini** [comment]
> Could you please rebase your PR @david-fortini ?

@finswimmer  It took some time, I think the PR should be ready now. Let me know if there are more remarks/ things to change.

**Comment by david-fortini** [comment]
@finswimmer Just a reminder ))

**Review by finswimmer — APPROVED** [review]
Thanks a lot. LGTM :+1:

**Comment by ddelange** [comment]
fwiw I added support for this pattern (e.g. google's infamous [`etils`](https://github.com/google/etils/blob/f03824f26402877b956729ddb7a5aba65c7a1f6b/pyproject.toml#L92) package which is required by `jax`) to pipgrip's vendored mixology in https://github.com/ddelange/pipgrip/pull/101

**Pull request merged**